### PR TITLE
AMM-871 Abdm Facility fetch services changes

### PR DESCRIPTION
**Pull request by helenKaryamsetty**
## 📋 Description

JIRA ID: AMM-871

Please provide a summary of the change and the motivation behind it. Include relevant context and details.

---

## ✅ Type of Change

- [ ] 🐞 **Bug fix** (non-breaking change which resolves an issue)
- [x] ✨ **New feature** (non-breaking change which adds functionality)
- [ ] 🔥 **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🛠 **Refactor** (change that is neither a fix nor a new feature)
- [ ] ⚙️ **Config change** (configuration file or build script updates)
- [ ] 📚 **Documentation** (updates to docs or readme)
- [ ] 🧪 **Tests** (adding new or updating existing tests)
- [ ] 🎨 **UI/UX** (changes that affect the user interface)
- [ ] 🚀 **Performance** (improves performance)
- [ ] 🧹 **Chore** (miscellaneous changes that don't modify src or test files)

---

## ℹ️ Additional Information

Please describe how the changes were tested, and include any relevant screenshots, logs, or other information that provides additional context.

## Summary by CodeRabbit

- **New Features**
	- Introduced new configuration properties for API endpoints, including `abdmFacilityId` and updated endpoints to version `v0.5`.
	- Added new methods in the `FacilityService` interface for fetching registered facilities and saving facility IDs.
	- Implemented a new `FacilityServiceImpl` class to handle facility-related operations.

- **Bug Fixes**
	- Enhanced error handling and logging in various service methods to improve traceability and prevent potential issues.

- **Refactor**
	- Updated data types from `Long` to `BigInteger` across multiple classes and repositories to accommodate larger values.
	- Streamlined code by integrating Lombok for automatic method generation, reducing boilerplate.

- **Documentation**
	- Updated method signatures and added comments to clarify changes related to new data types and configurations.

### CODE REVIEW

**Comment by drtechie** [comment]
@coderabbitai full review

File: src/main/java/com/wipro/fhir/controller/facility/FacilityController.java
Comment (drtechie) [inline src/main/java/com/wipro/fhir/controller/facility/FacilityController.java:1]: Indents are all over the place @helenKaryamsetty

File: src/main/java/com/wipro/fhir/data/mongo/amrit_resource/AMRIT_ResourceMongo.java [hunk 1]
```diff
@@ -65,68 +69,4 @@ public class AMRIT_ResourceMongo {
 	@Field(value = "CreateDate")
 	private Timestamp createDate = new Timestamp(System.currentTimeMillis());
 
-	public Long getBeneficiaryRegID() {
```
Comment (drtechie) [inline src/main/java/com/wipro/fhir/data/mongo/amrit_resource/AMRIT_ResourceMongo.java:68]: Several of these attributes are removed. Is this done on purpose?

Comment (helenKaryamsetty) [reply src/main/java/com/wipro/fhir/data/mongo/amrit_resource/AMRIT_ResourceMongo.java:68]: yes @drtechie its original declaration is "BigInteger" but given as "Long" all over... in upgradation(latest version) it is not accepting so changed/removed at almost all places.

File: src/main/java/com/wipro/fhir/data/mongo/care_context/SaveFacilityIdForVisit.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.wipro.fhir.data.mongo.care_context;
+
+import java.math.BigInteger;
+
+import lombok.Data;
+
+@Data
+public class SaveFacilityIdForVisit {
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/main/java/com/wipro/fhir/data/mongo/care_context/SaveFacilityIdForVisit.java:8]: That handling is given in other module.. not required here.

File: src/main/java/com/wipro/fhir/service/facility/FacilityService.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.wipro.fhir.service.facility;
+
+import com.wipro.fhir.utils.exception.FHIRException;
+
+public interface FacilityService {
+
+	String fetchRegisteredFacilities() throws FHIRException;
+
+	String saveAbdmFacilityId(String reqObj) throws FHIRException;
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/main/java/com/wipro/fhir/service/facility/FacilityService.java:9]: its not mandatory my response is fixed, so giving generic declaration as String

File: src/main/java/com/wipro/fhir/service/common/CommonServiceImpl.java [hunk 4]
```diff
@@ -310,63 +324,78 @@ public int addCareContextToMongo(PatientDemographic pDemo, PatientEligibleForRes
...
+							break;
+
+						case 3:
+							pcc.setGender("O");
+							break;
+
+						default:
+							break;
+						}
 					}
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/main/java/com/wipro/fhir/service/common/CommonServiceImpl.java:381]: As of now we don't intend to declare separately and use. So, using directly

File: src/main/java/com/wipro/fhir/service/common/CommonServiceImpl.java [hunk 5]
```diff
@@ -310,63 +324,78 @@
 			// get benid
 //			if (benRegID != null)
 //				benID = benHealthIDMappingRepo.getBenID(benRegID);
+			
+			// fetch abdm facility id  
+			logger.info("********t_benvisistData fetch request pvisit data :" ,  pVisit);
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/main/java/com/wipro/fhir/service/common/CommonServiceImpl.java:329]: there are no changes of pvisit to be null. so, not checking for the null condition

File: src/main/environment/common_ci.properties [hunk 6]
```diff
@@ -77,6 +81,10 @@ abdmGenerateMobileOTP=@env.ABDM_HEALTH_ID_BASE_URL@/api/v1/registration/aadhaar/
 abdmConfirmAadhaarBio=@env.ABDM_HEALTH_ID_BASE_URL@/api/v1/auth/confirmWithAadhaarBio
 abdmAccountProfile=@env.ABDM_Account_ID_BASE_URL@/api/v1/account/profile
 
+##ABDM Facility services
+getAbdmFacilityServicies= https://dev.abdm.gov.in/devservice/v1/bridges/getServices
```
Comment (drtechie) [inline src/main/environment/common_ci.properties:85]: CI shouldn't have this dev domain hardcoded. Use @env.ABDM_BASE_URL@?

**Review by ravishanigarapu — APPROVED** [review]
looks ok

**Pull request merged**